src/main/java/bio/terra/service/profile/ProfileDao.java

```diff
@@ -74,6 +75,28 @@ public BillingProfileModel createBillingProfile(BillingProfileRequestModel profi
             .createdDate(keyHolder.getTimestamp("created_date").toInstant().toString());
     }
 
+    @Transactional(propagation = Propagation.REQUIRED, isolation = Isolation.SERIALIZABLE)
+    public BillingProfileModel updateBillingProfileById(BillingProfileUpdateModel profileRequest) {
+        String sql = "UPDATE billing_profile "
+            + "SET billing_account_id = :billing_account_id, description = :description "
+            + "WHERE id = :id";
+        MapSqlParameterSource params = new MapSqlParameterSource()
+            .addValue("id", UUID.fromString(profileRequest.getId()))
+            .addValue("billing_account_id", profileRequest.getBillingAccountId())
+            .addValue("description", profileRequest.getDescription());
+        DaoKeyHolder keyHolder = new DaoKeyHolder();
+        jdbcTemplate.update(sql, params, keyHolder);
```

Comment:
Do we normally not check results from `update()`? I would expect something like 
```
        int numRows = jdbcTemplate.update(sql, params, keyHolder);
        if (numRows != 1) {
             throw ...;
        }
```
Unless we expect this to modify more than one row?

Reply:
We only do checks where the count is used for later logic. Not for the PK case

Reply:
I added a check here. As explained in the code comments, I'm throwing a "ProfileNotFoundException." I'd appreciate a sanity check on the assumptions I made here!